dz-9/src/main/java/org/example/Woman.java

```diff
@@ -0,0 +1,27 @@
+package org.example;
+public class Woman extends Person {
+    private String partnerLastName;
```

Comment:
May be better to use `initialLastName` for woman and set it in constructor, you can use this field in `deregisterPartnership` to revert lastName